paddlenlp/trainer/trainer.py

```diff
@@ -1688,10 +1704,11 @@ def _prepare_input(self, data: Union[paddle.Tensor, Any]) -> Union[paddle.Tensor
         elif isinstance(data, paddle.Tensor):
             # kwargs = dict(device=self.args.current_device)
             # update data type for pure fp16
-            if data.place.is_cuda_pinned_place():
-                return data.cuda()
-            return data
+            # if data.place.is_cuda_pinned_place():
+            #     return data.cuda()
+            # return data
             # return data.to(**kwargs)
+            return data._to(self.args.current_device, None, False)
```

Comment:
The `_to` api is in-place api and have serious bug in here. don't using it please.

Reply:
Done, discard this change.